classes/checkout/CheckoutSession.php

```diff
@@ -95,4 +95,9 @@ public function getGift()
             'message'   => $this->context->cart->gift_message
         ];
     }
+
+    public function isGuestAllowed()
```

Comment:
maybe, but I like having all the information I need for controlling the behaviour of the checkout in the same service (granted, `CheckoutSession` is perhaps not the best of names)